Software/src/wizard/ConfigureDevicePowerPage.cpp

```diff
@@ -0,0 +1,108 @@
+ *	by the Free Software Foundation, either version 2 of the License, or
+ *	(at your option) any later version.
+ *
+ *	Prismatik and Lightpack files is distributed in the hope that it will be
+ *	useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+ *	General Public License for more details.
+ *
+ *	You should have received a copy of the GNU General Public License
+ *	along with this program.	If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <QDesktopWidget>
+#include <QMessageBox>
+
+#include "ConfigureDevicePowerPage.hpp"
+#include "ui_ConfigureDevicePowerPage.h"
+#include "Settings.hpp"
+#include "LedDeviceDrgb.hpp"
+#include "LedDeviceDnrgb.hpp"
+#include "LedDeviceWarls.hpp"
+#include "Wizard.hpp"
+
+using namespace SettingsScope;
+
+ConfigureDevicePowerPage::ConfigureDevicePowerPage(bool isInitFromSettings, TransientSettings *ts, QWidget *parent):
+	QWizardPage(parent),
+	SettingsAwareTrait(isInitFromSettings, ts),
+	ui(new Ui::ConfigureDevicePowerPage)
+{
+	ui->setupUi(this);
+}
+
+void ConfigureDevicePowerPage::initializePage()
+{
+	SupportedDevices::DeviceType device = SupportedDevices::DefaultDeviceType;
+	if (field("isDrgb").toBool())
+		device = SupportedDevices::DeviceTypeDrgb;
+	else if (field("isDnrgb").toBool())
+		device = SupportedDevices::DeviceTypeDnrgb;
+	else if (field("isWarls").toBool())
+		device = SupportedDevices::DeviceTypeWarls;
+	else if (field("isLightpack").toBool())
+		device = SupportedDevices::DeviceTypeLightpack;
+	else if (field("isAdalight").toBool())
+		device = SupportedDevices::DeviceTypeAdalight;
+	else if (field("isArdulight").toBool())
+		device = SupportedDevices::DeviceTypeArdulight;
+	else if (field("isVirtual").toBool())
+		device = SupportedDevices::DeviceTypeVirtual;
+	else if (field("isAlienFx").toBool())
+		device = SupportedDevices::DeviceTypeAlienFx;
+
+	ui->sbLedMilliAmps->setValue(Settings::getDeviceLedMilliAmps(device));
+	ui->sbPowerSupplyAmps->setValue(Settings::getDevicePowerSupplyAmps(device));
+}
+
+
+bool ConfigureDevicePowerPage::validatePage()
+{
+	SupportedDevices::DeviceType device = SupportedDevices::DefaultDeviceType;
+	if (field("isDrgb").toBool())
+		device = SupportedDevices::DeviceTypeDrgb;
+	else if (field("isDnrgb").toBool())
+		device = SupportedDevices::DeviceTypeDnrgb;
+	else if (field("isWarls").toBool())
+		device = SupportedDevices::DeviceTypeWarls;
+	else if (field("isLightpack").toBool())
+		device = SupportedDevices::DeviceTypeLightpack;
+	else if (field("isAdalight").toBool())
+		device = SupportedDevices::DeviceTypeAdalight;
+	else if (field("isArdulight").toBool())
+		device = SupportedDevices::DeviceTypeArdulight;
+	else if (field("isVirtual").toBool())
+		device = SupportedDevices::DeviceTypeVirtual;
+	else if (field("isAlienFx").toBool())
+		device = SupportedDevices::DeviceTypeAlienFx;
```

Comment:
Consider putting this block in a helper function (for readability and to avoid going out of sync)

Reply:
TBH I don't like the current way of handling device types and grabbers. I'm sure it was fine at the start, but as both lists grow you have to duplicate tons of stuff all over the place and have this kind of redundancies.
Since the addition of UDP stuff I wanted to take a look at that, probably before multidevice thing. So I'll leave that for a future PR.
If you already have some ideas on both subjects I'm all ears.

Reply:
Some issues can be solved through the inheritance (as already is the case) but especially at the intersection with the UI some manual switching/selecting will be necessary I think